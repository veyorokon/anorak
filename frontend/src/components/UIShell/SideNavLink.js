/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { settings } from 'carbon-components';
import cx from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import Link, { LinkPropTypes } from './Link';
import SideNavIcon from './SideNavIcon';
import SideNavItem from './SideNavItem';
import SideNavLinkText from './SideNavLinkText';

const { prefix } = settings;

const SideNavLink = ({
  className: customClassName,
  children,
  icon,
  isActive,
  ...rest
}) => {
  const className = cx({
    [`${prefix}--side-nav__link`]: true,
    [`${prefix}--side-nav__link--current`]: isActive,
    [customClassName]: !!customClassName,
  });

  return (
    <SideNavItem>
      <Link {...rest} className={className}>
        <SideNavIcon small>{icon}</SideNavIcon>
        <SideNavLinkText>{children}</SideNavLinkText>
      </Link>
    </SideNavItem>
  );
};

SideNavLink.propTypes = {
  ...LinkPropTypes,

  /**
   * Provide an optional class to be applied to the containing node
   */
  className: PropTypes.string,

  /**
   * Specify the text content for the link
   */
  children: PropTypes.string.isRequired,
};

SideNavLink.defaultProps = {
  element: 'a',
};

export const createCustomSideNavLink = element => props => {
  return <SideNavLink element={element} {...props} />;
};

export default SideNavLink;
