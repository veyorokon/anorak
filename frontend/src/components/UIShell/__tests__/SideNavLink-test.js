/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import SideNavLink from '../SideNavLink';

describe('SideNavLink', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      className: 'custom-classname',
      children: 'link',
      icon: <div>icon</div>,
      isActive: false,
    };
  });

  it('should render', () => {
    const inactive = mount(<SideNavLink {...mockProps} />);
    expect(inactive).toMatchSnapshot();
    const active = mount(<SideNavLink {...mockProps} isActive />);
    expect(active).toMatchSnapshot();
  });
});
