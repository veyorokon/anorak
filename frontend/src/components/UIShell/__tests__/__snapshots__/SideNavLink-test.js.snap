// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavLink should render 1`] = `
<SideNavLink
  className="custom-classname"
  element="a"
  icon={
    <div>
      icon
    </div>
  }
  isActive={false}
>
  <SideNavItem>
    <li
      className="bx--side-nav__item"
    >
      <Link
        className="bx--side-nav__link custom-classname"
        element="a"
      >
        <a
          className="bx--side-nav__link custom-classname"
        >
          <SideNavIcon
            small={true}
          >
            <div
              className="bx--side-nav__icon bx--side-nav__icon--small"
            >
              <div>
                icon
              </div>
            </div>
          </SideNavIcon>
          <SideNavLinkText>
            <span
              className="bx--side-nav__link-text"
            >
              link
            </span>
          </SideNavLinkText>
        </a>
      </Link>
    </li>
  </SideNavItem>
</SideNavLink>
`;

exports[`SideNavLink should render 2`] = `
<SideNavLink
  className="custom-classname"
  element="a"
  icon={
    <div>
      icon
    </div>
  }
  isActive={true}
>
  <SideNavItem>
    <li
      className="bx--side-nav__item"
    >
      <Link
        className="bx--side-nav__link bx--side-nav__link--current custom-classname"
        element="a"
      >
        <a
          className="bx--side-nav__link bx--side-nav__link--current custom-classname"
        >
          <SideNavIcon
            small={true}
          >
            <div
              className="bx--side-nav__icon bx--side-nav__icon--small"
            >
              <div>
                icon
              </div>
            </div>
          </SideNavIcon>
          <SideNavLinkText>
            <span
              className="bx--side-nav__link-text"
            >
              link
            </span>
          </SideNavLinkText>
        </a>
      </Link>
    </li>
  </SideNavItem>
</SideNavLink>
`;
